### PR TITLE
Fix flaky tests 'testDeleteAllAndListAllPits' and 'testDeleteWhileSearch'

### PR DESCRIPTION
**Pull request by bharath-techie**
### Description
testDeleteAllAndListAllPits : 
List all PITs and delete all PITs both were triggered asynchronously and test was flaky in times where delete all finishes before get all. So fixed the test to order these two calls ( get will always finish first and then delete will be triggered )

testDeleteWhileSearch:
Added check for an error message which is also one of the valid scenarios

Tested using :

```
./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.opensearch.client.PitIT.testDeleteAllAndListAllPits" -Dtests.iters=1000

 ./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.pit.DeletePitMultiNodeIT.testDeleteWhileSearch" -Dtests.iters=1000
```

### Related Issues

https://github.com/opensearch-project/OpenSearch/issues/5030
https://github.com/opensearch-project/OpenSearch/issues/9048

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented. [Not applicable]
  - [x] New functionality has javadoc added 
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))  [Not applicable]
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)  [Not applicable]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Arpit-Bandejiya — APPROVED** [review]

**Review by ajaymovva — APPROVED** [review]

**Comment by peternied** [comment]
> Failing checks are inspected and point to the corresponding known issue

@bharath-techie It looks like there were many failing checks - can you please inspect and create/update issues before this is merged?

**Comment by bharath-techie** [comment]
Gradle check failures known issues seen in this PR checks :
https://github.com/opensearch-project/OpenSearch/issues/9499 
https://github.com/opensearch-project/OpenSearch/issues/10755
https://github.com/opensearch-project/OpenSearch/issues/10945
https://github.com/opensearch-project/OpenSearch/issues/10902
https://github.com/opensearch-project/OpenSearch/issues/10154

Created issues :
https://github.com/opensearch-project/OpenSearch/issues/10972 - RemoteStoreIT flaky test

**Comment by bharath-techie** [comment]
Can someone help merging this PR ?

**Comment by bharath-techie** [comment]
@kotwanikunal @peternied  can you help review / merge the code if the fix looks good ?

**Comment by peternied** [comment]
> Gradle check failures known issues seen in this PR checks :
> ...

@bharath-techie Thank you very much!

File: client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java [hunk 1]
```diff
@@ -126,11 +124,12 @@ public void onFailure(Exception e) {
...
-        assertTrue(countDownLatch.await(10, TimeUnit.SECONDS));
+
         // validate no pits case
-        getAllPitResponse = highLevelClient().getAllPits(RequestOptions.DEFAULT);
-        assertTrue(getAllPitResponse.getPitInfos().size() == 0);
-        highLevelClient().deleteAllPitsAsync(RequestOptions.DEFAULT, deletePitListener);
+        assertBusy(() -> {
+            GetAllPitNodesResponse getAllPitResponse1 = highLevelClient().getAllPits(RequestOptions.DEFAULT);
+            assertTrue(getAllPitResponse1.getPitInfos().size() == 0);
+            highLevelClient().deleteAllPitsAsync(RequestOptions.DEFAULT, deletePitListener);
```
Comment (peternied) [inline client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java:132]: What was wrong with the previous single collection + delete, doesn't `deleteAllPitsAsync` have a deterministic conclusion, shouldn't all the Pit be deleted after only one call?

This  smells like there might be an implementation bug, can you help me understand?

Comment (bharath-techie) [reply client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java:132]: no there is a race condition between get and delete, so if delete finishes first , get pits assert fails leading to flaky tests

Comment (bharath-techie) [reply client/rest-high-level/src/test/java/org/opensearch/client/PitIT.java:132]: Also another fix is i removed countdown latch with assertbusy , both are technically the same as wait time is at max 10 seconds - this was not needed but did it to make the code look clean.

**Review by peternied — APPROVED** [review]
Thanks for addressing this flaky test @bharath-techie

**Pull request merged**